src/server/src/main/java/io/cassandrareaper/service/SegmentGenerator.java

```diff
@@ -29,6 +35,8 @@
 final class SegmentGenerator {
 
   private static final Logger LOG = LoggerFactory.getLogger(SegmentGenerator.class);
+  private static final boolean COALESCING_DISABLED =
+      Boolean.getBoolean("disable.tokenrange.coalescing");
```

Comment:
nit-pick: the operator should carry. (otherwise it gives a false impression that arguments/parameters are being carried.) this applies generally to your patches.

Reply:
nit-pick: i'd favour prefixing system variables with an appropriate application label so to provide some context/taxonomy.
for example ` SegmentGenerator.class.getName() + ".coalescing.disabled"` 

(system variables can come from anywhere, just not the application, so it nice to isolate them appropriately, especially a flag that's not to be generally announced.)